### PR TITLE
feat: introduce content modified timestamp

### PR DESCRIPTION
**Pull request by iSCJT**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Implements a _lastEditTimestamp property to the document metadata which is only updated when a 'real' update/edit happens on a document.  The platform was using the modified date but as this is updated when someone comments or votes useful etc. it was leading to incorrect/misleading last edit/update dates being shown when viewing how-to's and research.

## Git Issues

Closes #

## Screenshots/Videos

_If useful, provide screenshot or capture to highlight main changes_

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: functions/src/Integrations/firebase-discord.ts [hunk 1]
```diff
@@ -51,9 +51,13 @@ export const notifyEventAccepted = functions.firestore
...
+    const prevInfo = change.before.exists ? change.before.data() : null
+    const previouslyAccepted = prevInfo?.moderation === 'accepted'
+    const shouldNotify = info.moderation === 'accepted' && !previouslyAccepted
+    if (!shouldNotify) {
       return null
     }
+
     const user = info._createdBy
     const url = info.url
     const location = info.location.country
```
Comment (thisislawatts) [inline functions/src/Integrations/firebase-discord.ts:63]: **question:** This seems unrelated. Could it be moved into a standalone PR?

File: src/stores/databaseV2/DocReference.tsx [hunk 2]
```diff
@@ -120,10 +125,14 @@ export class DocReference<T> {
       _modified: modifiedTimestamp,
     }
 
+    if (o.set_last_edit_timestamp)
+      meta._lastEditTimestamp = new Date().toISOString()
```
Comment (thisislawatts) [inline src/stores/databaseV2/DocReference.tsx:129]: **issue:** Consider using `_contentModifiedTimestamp` as the property name. It removes some of the ambiguity about what is being recorded here. I worry that `_lastEditTimestamp` is still a little too vague.

Comment (iSCJT) [reply src/stores/databaseV2/DocReference.tsx:129]: That makes sense to me, I'll refactor.

File: functions/src/Integrations/firebase-discord.ts (hunk 1 above)
Comment (iSCJT) [reply functions/src/Integrations/firebase-discord.ts:60]: The function was erroring as a knock on from the migration to update the event records with the the content modified timestamp.  I've removed the changes and added #2431

File: src/stores/databaseV2/DocReference.tsx (hunk 2 above)
Comment (iSCJT) [reply src/stores/databaseV2/DocReference.tsx]: @thisislawatts I've finished the refactor now.

**Review by thisislawatts — APPROVED** [review]
